src/GitVersion.Core.Tests/IntegrationTests/ContinuousDeploymentTestScenarios.cs

```diff
@@ -0,0 +1,405 @@
+
+        fixture.BranchTo("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-just-a-test.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-just-a-test.1", configuration);
+
+        fixture.Checkout("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.0", configuration);
+
+        fixture.MergeNoFF("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configuration);
+
+        fixture.Repository.Branches.Remove("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configuration);
+
+        fixture.Repository.DumpGraph();
+    }
+
+    [Test]
+    public void ShouldFallbackToTheVersionOnDevelopLikeTheReleaseWasNeverCreatedWhenReleaseHasBeenCanceled()
+    {
+        // *8f062c7 49 minutes ago(HEAD -> develop)
+        // |\  
+        // | *bda6ba8 52 minutes ago
+        // | *6f5cf19 54 minutes ago
+        // * | 3b20f15 50 minutes ago
+        // |/
+        // *f5640b3 56 minutes ago
+        // *2099a07 58 minutes ago(main)
+
+        var configuration = TestConfigurationBuilder.New
+            .WithVersioningMode(VersioningMode.ContinuousDeployment)
+            .WithoutAnyTrackMergeTargets().Build();
+
+        using var fixture = new EmptyRepositoryFixture();
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.0.1-ci.1", configuration);
+
+        fixture.BranchTo("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.2", configuration);
+
+        fixture.BranchTo("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configuration);
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configuration);
+
+        fixture.MergeNoFF("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.4", configuration);
+
+        // cancel the release 1.0.0
+        fixture.Repository.Branches.Remove("release/1.0.0");
+
+        // ❌ expected: "0.1.0-alpha.6"
+        fixture.AssertFullSemver("1.1.0-alpha.4", configuration);
```

Comment:
At this moment the assertion is correct. But in my opinion some user can see it different. That means we need to discuss it and make it configurable to fit all needs.

Reply:
Then please remove or change the comment and create an issue where we can discuss it instead. :)

Reply:
Okay I'm ensure that we have existing issues for that to address it. If it is okay with you I would like to used a question mark as comment to document that this might be changing. :) // ❔